src/operation.js

```diff
@@ -24,26 +24,34 @@ export default (context, name, operationName) => async (req, args, options) => {
         operationName,
     });
 
-    return http(
-        buildRequest(
-            requestContext,
-            req,
-            args,
-            options,
-        ),
-    ).then(
-        response => buildResponse(requestContext)(
-            response,
-            requestContext,
-            req,
-            options,
-        ),
-    ).catch(
-        error => buildError(requestContext)(
-            error,
-            requestContext,
-            req,
-            options,
-        ),
+    const request = buildRequest(
+        requestContext,
+        req,
+        args,
+        options,
+    );
+    const retries = get(request, 'retries');
+
+    let errorResponse;
+    for (let attempts = 0; attempts < retries; attempts++) {
+        try {
+            /* eslint-disable no-await-in-loop */
```

Comment:
Await-in-loop is mostly to prevent you from doing something like:
```
for item in items:
   await item
```
Here I'm just retrying the same event repeatedly.